src/renderer/api-decorator.js

```diff
@@ -524,18 +524,31 @@ limitations under the License.
      * Preload script eval
      */
     ipc.once(`post-api-injection-${renderFrameId}`, () => {
-        const preload = windowOptions.preload;
+        const preloads = windowOptions.preload;
+        let preloadPayloads;
 
-        if (preload) {
-            try {
-                const preloadScriptContent = syncApiCall('get-window-preload-script', {
-                    preload
-                });
-                window.eval(preloadScriptContent); /* jshint ignore:line */
-            } catch (err) {
-                console.error(err);
-            }
+        try {
+            preloadPayloads = syncApiCall('get-window-preload-scripts', { preloads });
+        } catch (err) {
+            console.error(err);
+            return;
         }
+
+        // `preloadPayloads` or `preloadPayload` are locally accessible to eval'd scripts. DO NOT RENAME!
```

Comment:
Quick personal opinion without thinking this through: preload scripts should have **no** access or **rely** on any variables in the scope they are eval'ed